src/ContractInteractor.ts

```diff
@@ -877,6 +892,22 @@ export default class ContractInteractor {
             `No receipt found for this transaction ${transactionHash}`
         );
     }
+
+    async getERC20Token(address: string): Promise<Token> {
+        const token = await this._createERC20(address);
+        const [name, decimals, symbol] = await Promise.all([
+            await token.name(),
+            (await token.decimals()).toNumber(),
+            await token.symbol()
+        ]);
+
+        return {
+            name,
+            decimals,
+            symbol,
+            contractAddress: address
+        };
```

Comment:
Do we need to wait for all those properties? I'd let the user uses whatever property they want instead of retrieving all of them here.

Reply:
updated